src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +279,837 @@ static DamageTable D_808792D4 = {
+                    this->collider1.elements[arg2].info.bumper.hitPos.y,
+                    this->collider1.elements[arg2].info.bumper.hitPos.z, 0, 0, 0, CLEAR_TAG_LARGE_LIGHT_RAYS);
+    }
+}
+
+void func_80876CAC(EnDodongo* this) {
+    this->unk_300 = 10;
+    this->collider1.base.colType = 3;
+    this->unk_340 = 0.75f;
+    this->unk_344 = 1.125f;
+    this->unk_33C = 1.0f;
+    this->timer = 80;
+    this->actor.flags &= ~0x400;
+    func_800BCB70(&this->actor, 0x4000, 0xFF, 0, 80);
+}
+
+void func_80876D28(EnDodongo* this, GlobalContext* globalCtx) {
+    if (this->unk_300 == 10) {
+        this->timer = 0;
+        this->actor.colorFilterTimer = 0;
+        this->unk_300 = 0;
+        this->collider1.base.colType = 0;
+        this->unk_33C = 0.0f;
+        func_800BF7CC(globalCtx, &this->actor, &this->unk_348[0], 9, 2, this->unk_334 * 0.3f, this->unk_334 * 0.2f);
+        this->actor.flags |= 0x400;
+    }
+}
+
+void func_80876DC4(EnDodongo* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f sp80;
+    Vec3f sp74;
+    Vec3f sp68;
+    s16 sp66;
+    s16 sp64;
+    s16 sp62;
+    f32 temp_f22;
+    f32 temp_f20;
+
+    Math_Vec3f_Copy(&sp68, &this->unk_348[0]);
+    sp66 = (Rand_Next() >> 0x12) + this->actor.shape.rot.y;
+    temp_f20 = Math_CosS(sp66);
+    temp_f22 = Math_SinS(sp66);
+
+    if (this->unk_32C.r > 30) {
+        this->unk_32C.r -= 16;
+        this->unk_32C.g += -0x10;
+    }
+
+    if (this->unk_32C.b < 30) {
+        this->unk_32C.b += 5;
+        this->unk_32C.a += 8;
+        this->unk_330.a += 8;
+    }
+
+    if (this->unk_330.r != 0) {
+        this->unk_330.r -= 15;
+    }
+
+    if (this->unk_330.g != 0) {
+        this->unk_330.g--;
+    }
+
+    sp80.x = temp_f22 * 3.5f * this->unk_334;
+    sp80.y = this->unk_330.r * 0.02f * this->unk_334;
+    sp80.z = temp_f20 * 3.5f * this->unk_334;
+    sp74.x = (Rand_ZeroFloat(0.1f) + 0.15f) * -temp_f22 * this->unk_334;
+    sp74.y = this->unk_334 * 0.3f;
+    sp74.z = (Rand_ZeroFloat(0.1f) + 0.15f) * -temp_f20 * this->unk_334;
+    sp64 = this->unk_334 * 100.0f;
+    sp62 = this->unk_334 * 25.0f;
+    func_800B0EB0(globalCtx, &sp68, &sp80, &sp74, &this->unk_32C, &this->unk_330, sp64, sp62, 0x14);
+
+    sp66 = ((u32)Rand_Next() >> 0x13) + this->actor.shape.rot.y;
+    temp_f20 = Math_CosS(sp66);
+    temp_f22 = Math_SinS(sp66);
+
+    sp68.x -= (temp_f20 * 6.0f) * this->unk_334;
+    sp68.z += (temp_f22 * 6.0f) * this->unk_334;
+
+    sp80.x = -temp_f20 * 3.5f * this->unk_334;
+    sp80.z = temp_f22 * 3.5f * this->unk_334;
+
+    sp74.x = (Rand_ZeroFloat(0.1f) + 0.15f) * temp_f20 * this->unk_334;
+    sp74.z = (Rand_ZeroFloat(0.1f) + 0.15f) * -temp_f22 * this->unk_334;
+    func_800B0EB0(globalCtx, &sp68, &sp80, &sp74, &this->unk_32C, &this->unk_330, sp64, sp62, 0x14);
+
+    sp68.x = this->unk_348[0].x + ((temp_f20 * 6.0f) * this->unk_334);
+    sp68.z = this->unk_348[0].z - (temp_f22 * 6.0f * this->unk_334);
```

Comment:
```suggestion
    sp68.x = this->unk_348[0].x + (temp_f20 * 6.0f * this->unk_334);
    sp68.z = this->unk_348[0].z - (temp_f22 * 6.0f * this->unk_334);
```
Just so `x` and `z` are similar